core/lib/Cardano/Address/Style/Icarus.hs

```diff
@@ -311,16 +315,20 @@ deriveAddressPublicKey =
 -- > "addr1vxpfffuj3zkp5g7ct6h4va89caxx9ayq2gvkyfvww48sdncxsce5t"
 --
 -- | Analyze an 'Address' to know whether it's an Icarus address or not.
--- Returns 'Nothing' if the address isn't a byron address, or return a
+-- Throws 'AddrError' if the address isn't a byron address, or return a
 -- structured JSON that gives information about an address.
 --
 -- @since 2.0.0
-inspectIcarusAddress :: Address -> Maybe Json.Value
+inspectIcarusAddress :: MonadThrow m => Address -> m Json.Value
 inspectIcarusAddress addr = do
-    payload <- CBOR.deserialiseCbor CBOR.decodeAddressPayload bytes
-    (root, attrs) <- CBOR.deserialiseCbor decodePayload payload
-    guard $ 1 `notElem` (fst <$> attrs)
-    ntwrk <- CBOR.deserialiseCbor CBOR.decodeProtocolMagicAttr payload
+    payload <- either (throwM . DeserialiseError addr) pure
+            $ CBOR.deserialiseCbor CBOR.decodeAddressPayload bytes
+    (root, attrs) <- either (throwM . DeserialiseError addr) pure
+                  $ CBOR.deserialiseCbor decodePayload payload
+    _  <- maybe (throwM $ InvalidDerivationPath addr) pure
+       $ find ((== 1) . fst) attrs
```

Comment:
This actually changes the semantic of the function. Before, this was checking that there was no derivation path present in the address attributes (that's how we distinguish legacy Byron and Icarus addresses). Now, it does the opposite.